### PR TITLE
dev-libs/libqt5pas: new package, add 1.2.9

### PR DESCRIPTION
**Pull request by corvus1**
Needed for switching dev-lang/lazarus to Qt5 GUI. ( https://github.com/gentoo/gentoo/pull/28622 )

UPD: This PR is merged into https://github.com/gentoo/gentoo/pull/28622 
Please direct any further comments and reviews there. 

Signed-off-by: Michael Corvinus <voron1@gmail.com>

### CODE REVIEW

**Comment by scardracs** [comment]
You forgot to add you mail and legal name

**Comment by corvus1** [comment]
> You forgot to add you mail and legal name.

I assume you mean email, it's right there, in the commit message. As to legal name, it's also right there. It is my legal name.  Or did I forget to add it somewhere else?

> EDIT: readd sourceforge and merge commits: one for the ebuild and one for the metadata

Not sure how to merge commits properly. As a git user I am barely out of my diapers.

**Comment by corvus1** [comment]
I went rogue, removed the whole thing and re-added. I feel like I am making `git mess` out of this one.

File: dev-libs/libqt5pas/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        <email>proxy-maint@gentoo.org</email>
+        <name>Proxy Maintainers</name>
+    </maintainer>
+    <upstream>
+        <remote-id type="gitlab">freepascal.org/lazarus</remote-id>
+    </upstream>
+    <upstream>
+        <remote-id type="sourceforge">lazarus</remote-id>
+    </upstream>
+</pkgmetadata>
```
Comment (scardracs) [inline dev-libs/libqt5pas/metadata.xml:14]:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
<pkgmetadata>
    <maintainer type="person" proxied="yes">
        <email>voron1@gmail.com</email>
        <name>Michael Corvinus</name>
    </maintainer>
    <maintainer type="project" proxied="proxy">
        <email>proxy-maint@gentoo.org</email>
        <name>Proxy Maintainers</name>
    </maintainer>
    <upstream>
        <remote-id type="gitlab">freepascal.org/lazarus</remote-id>
        <remote-id type="sourceforge">lazarus</remote-id>
    </upstream>
</pkgmetadata>
```

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+LAZRELEASE="2.2.4"
+
+if [[ ${PV} = 9999* ]]; then
+	EGIT_REPO_URI="https://gitlab.com/freepascal.org/lazarus/lazarus.git"
+	inherit git-r3
+else
+	SRC_URI="https://sourceforge.net/projects/lazarus/files/Lazarus%20Zip%20_%20GZip/Lazarus%20${LAZRELEASE}/lazarus-${LAZRELEASE}-0.tar.gz/download -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
```
Comment (scardracs) [inline dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild:16]: Detete everything and leave only
```
SRC_URI="https://sourceforge.net/projects/lazarus/files/Lazarus%20Zip%20_%20GZip/Lazarus%20${LAZRELEASE}/lazarus-${LAZRELEASE}-0.tar.gz/download -> ${P}.tar.gz"
KEYWORDS="~amd64 ~x86"
```
as you have not a live ebuild too

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0/2.2"
+
+DEPEND="
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtprintsupport:5
+	dev-qt/qtx11extras:5
+"
+RDEPEND="${DEPEND}"
+BDEPEND=""
```
Comment (scardracs) [inline dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild:30]: Delete

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+S="${WORKDIR}/lazarus/lcl/interfaces/qt5/cbindings"
+
+src_unpack () {
+	if [[ ${PV} = 9999* ]]; then
+		git-r3_fetch ${EGIT_REPO_URI}
+		git-r3_checkout ${EGIT_REPO_URI} "${WORKDIR}/lazarus" "" "lcl/interfaces/qt5/cbindings"
+	else
```
Comment (scardracs) [inline dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild:38]: Same, delete as you aren't using a live ebuild

**Comment by scardracs** [comment]
Don't worry, you can go inside your branch and use git rebase -i HEAD~10 (it will show the last 10 commits). You can the use f for fixup, d for discard and r for reword. After that use pkgdev push to push upstream new commits

File: dev-libs/libqt5pas/metadata.xml (hunk 1 above)
Comment (corvus1) [reply dev-libs/libqt5pas/metadata.xml]: Done.

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild (hunk 2 above)
Comment (corvus1) [reply dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild]: I was hoping to use it in the next phase, but done.

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+if [[ ${PV} = 9999* ]]; then
+	EGIT_REPO_URI="https://gitlab.com/freepascal.org/lazarus/lazarus.git"
+	inherit git-r3
+else
+	SRC_URI="https://sourceforge.net/projects/lazarus/files/Lazarus%20Zip%20_%20GZip/Lazarus%20${LAZRELEASE}/lazarus-${LAZRELEASE}-0.tar.gz/download -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Free Pascal Qt5 bindings library updated by lazarus IDE."
+HOMEPAGE="https://gitlab.com/freepascal.org/lazarus/lazarus"
```
[the remark this answers is not in the document]
Comment (corvus1) [reply dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild:11]: Done.

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild (hunk 3 above)
Comment (corvus1) [reply dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild]: Deleted.

File: dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild (hunk 4 above)
Comment (corvus1) [reply dev-libs/libqt5pas/libqt5pas-1.2.9.ebuild]: I should probably remove src_unpack() altogether then, no?

**Comment by corvus1** [comment]
Oh, nice. The CI is happy now.

**Review by scardracs — APPROVED** [review]
LGTM

**Comment by scardracs** [comment]
Just a little point. I've run your PR in a test bench and it turns out that there are 2 possible missing deps.

```
 * CMP: listing dev-libs/libqt5pas-1.2.9/image info
 * SONAME: libQt5Pas.so.1(64)
 * VDB: detected possibly incorrect RDEPEND (dev-libs/libqt5pas-1.2.9)
 *  > dev-qt/qtcore:5
 *  > dev-qt/qtwidgets:5

```

**Comment by corvus1** [comment]
Done.
I think I more or less figured out how to squash and split all this into something decent. Do I need to do that?

**Comment by scardracs** [comment]
I see you moved the add of libqt5pas on the other PR so you can close that one and maintain only the other :)

**Pull request closed without merging**